pontoon/base/models.py

```diff
@@ -25,12 +25,69 @@
 log = logging.getLogger('pontoon')
 
 
+class UserTranslationsManager(models.Manager):
+    """
+    Provides various method to interact with larger sets of translations and their stats for user.
+    """
+
+    def _changed_translations_count(self, *args):
+        """
+        Helper method, returns expression object which allows us to annotate querysets
+        with counts of translations.
+        """
+        translation_query = (
+            ~Q(translation__string=F('translation__entity__string'))
+            & ~Q(translation__string=F('translation__entity__string'))
+        )
+
+        if args:
+            translation_query &= Q(*args)
+
+        # For each translation that matches the filter, return 1. Aggregate
+        # the sum of all those results to count the number of matching
+        # translations.
+        return Sum(
+            Case(
+                When(translation_query, then=1), output_field=models.IntegerField(), default=0))
+
+    def with_translation_counts(self, start_date=None, limit=100):
+        """
+        Returns contributors list, sorted by count of their translations.
+        Every user instance has added following properties:
+        * translations_count
+        * translations_approved_count
+        * translations_unapproved_count
+        * translations_needs_work_count
+        Method has been created mainly to improve performance and to optimize
+        count of sql queries during generation of metrics.
+        All counts will be returned from start_date to now().
+        :param date start_date: start date for translations.
+        """
+        def translations_count(query=None):
+            """Short helper to avoid duplication of passing dates."""
+            query = query or Q()
+            query &= Q(translation__date__gte=start_date)
```

Comment:
`start_date` is an optional argument, you should check that it's not `None` before using it here.